rspace/src/main/scala/coop/rchain/rspace/LMDBStore.scala

```diff
@@ -285,13 +297,33 @@ class LMDBStore[C, P, A, K] private (env: Env[ByteBuffer],
         }.toMap
       }
     }
+
+  private[rspace] def putTrie(txn: Txn[ByteBuffer],
+                              key: Blake2b256Hash,
+                              value: Trie[Blake2b256Hash, GNAT[C, P, A, K]]): Unit = {
+    val encodedValue = codecHistory.encode(value).get
+    val keyBuff      = toByteBuffer(key.bytes)
+    val valBuff      = toByteBuffer(encodedValue)
+    _dbTrie.put(txn, keyBuff, valBuff)
+  }
+
+  private[rspace] def getTrie(
+      txn: Txn[ByteBuffer],
+      key: Blake2b256Hash): Option[Trie[Blake2b256Hash, GNAT[C, P, A, K]]] = {
+    val keyBuff = toByteBuffer(key.bytes)
+    Option(_dbTrie.get(txn, keyBuff)).map { (buffer: ByteBuffer) =>
+      // ht: Yes, I want to throw an exception if deserialization fails
```

Comment:
Throwing is usually discouraged. Is there a good reason for doing this here?

Reply:
I don't want to poke my nose into Henry's code but I think that this is the kind of error for which there's no _graceful recover_ . Failing to deserialize our own trie means that we screw sth really bad on either ends of the serialization chain. It's safer to not allow for the app run than run in corrupted state.

Reply:
Fair enough!

Reply:
> I don't want to poke my nose into Henry's code

Nonsense!  All pokes are welcome!

Indeed, currently we are using exceptions for truly *exceptional* situations where core invariants are violated or from which we cannot recover.  Errors from LMDB and serialization/deserialization errors fall into this latter category.